src/ape/pytest/runners.py

```diff
@@ -137,34 +137,10 @@ def pytest_runtest_setup(self, item):
             or (hasattr(pytest, "DoctestItem") and isinstance(item, pytest.DoctestItem))
             or "_function_isolation" in item.fixturenames  # prevent double injection
         ):
-            # isolation is disabled via cmdline option
+            # isolation is disabled via cmdline option or running doc-tests.
             return
 
-        fixture_map = item.session._fixturemanager._arg2fixturedefs
-        scopes = [
-            definition.scope
-            for name, definitions in fixture_map.items()
-            if name in item.fixturenames
-            for definition in definitions
-        ]
-
-        for scope in ["session", "package", "module", "class"]:
-            # iterate through scope levels and insert the isolation fixture
-            # prior to the first fixture with that scope
-            try:
-                idx = scopes.index(scope)  # will raise ValueError if `scope` not found
-                item.fixturenames.insert(idx, f"_{scope}_isolation")
```

Comment:
I am not sure if this was the problem or not, but using the `idx` of the scope when inserting the fixturename seems wrong. A simple `append` does the trick (fixed below)

Reply:
Yes - doing more research, this seems like it was a problem (though still not sure if it is _the_ problem).
But basically you can end up in a case where the fixtures were not added in the right spot because it uses the first index of the scope and not the last.

I think having the isolation fixtures all at the end of the list makes sense, but going to experiment more to see.